src/core/globals/blockchain.c

```diff
@@ -0,0 +1,68 @@
+#include "blockchain.h"
+
+void blockchain_init() {
+  Block* genesis_block;
```

Comment:
Question on genesis block. Isn't it actually mined....aka the nonce is set such that it can be verified. Don't we want it's hash to match the difficulty pattern? or do we just ignore for genesis block?

Reply:
This should only matter if we implement verifying the entire blockchain from the start, and even then we can just skip the first block. Otherwise, as every node will ship with the genesis block already on the chain it shouldn't ever need to be validated. Might be worth finding the correct nonce once we have things running, but I think is alright for now?